core/providers/huggingface/huggingface.go

```diff
@@ -0,0 +1,990 @@
+			chunkIndex++
+
+			// Handle conversion to ResponsesStream if needed
+			if isResponsesToChatCompletionsFallback {
+				// Workaround for combined usage and content in the same chunk:
+				// If we have both usage and choices, split them into two separate processing steps
+				// to ensure content events are generated before completion events.
+				if response.Usage != nil && len(response.Choices) > 0 {
+					// 1. Process content without usage
+					contentResponse := *response
+					contentResponse.Usage = nil
+
+					responsesResponses := contentResponse.ToBifrostResponsesStreamResponse(responsesStreamState)
+					for _, responsesResp := range responsesResponses {
+						if responsesResp != nil {
+							responsesResp.ExtraFields.RequestType = schemas.ResponsesStreamRequest
+							bifrostStream := providerUtils.GetBifrostResponseForStreamResponse(nil, nil, responsesResp, nil, nil)
+							providerUtils.ProcessAndSendResponse(ctx, postHookRunner, bifrostStream, responseChan)
+						}
+					}
+
+					// 2. Process usage without choices (to trigger completion)
+					usageResponse := *response
+					usageResponse.Choices = nil
+
+					responsesResponses = usageResponse.ToBifrostResponsesStreamResponse(responsesStreamState)
+					for _, responsesResp := range responsesResponses {
+						if responsesResp != nil {
+							responsesResp.ExtraFields.RequestType = schemas.ResponsesStreamRequest
+							bifrostStream := providerUtils.GetBifrostResponseForStreamResponse(nil, nil, responsesResp, nil, nil)
+							providerUtils.ProcessAndSendResponse(ctx, postHookRunner, bifrostStream, responseChan)
+						}
+					}
+				} else {
+					// Convert chat streaming response to responses streaming responses
+					responsesResponses := response.ToBifrostResponsesStreamResponse(responsesStreamState)
+
+					for _, responsesResp := range responsesResponses {
+						if responsesResp != nil {
+							// Update RequestType for responses stream
+							responsesResp.ExtraFields.RequestType = schemas.ResponsesStreamRequest
+							bifrostStream := providerUtils.GetBifrostResponseForStreamResponse(nil, nil, responsesResp, nil, nil)
+							providerUtils.ProcessAndSendResponse(ctx, postHookRunner, bifrostStream, responseChan)
+						}
+					}
+				}
+			} else {
+				bifrostStream := providerUtils.GetBifrostResponseForStreamResponse(nil, response, nil, nil, nil)
+				providerUtils.ProcessAndSendResponse(ctx, postHookRunner, bifrostStream, responseChan)
+			}
+		}
+
+		if err := scanner.Err(); err != nil {
+			provider.logger.Warn(fmt.Sprintf("Error reading stream: %v", err))
+			providerUtils.ProcessAndSendError(ctx, postHookRunner, err, responseChan, schemas.ChatCompletionStreamRequest, providerName, request.Model, provider.logger)
+		}
+	}()
+
+	return responseChan, nil
+}
+
+func (provider *HuggingFaceProvider) Responses(ctx context.Context, key schemas.Key, request *schemas.BifrostResponsesRequest) (*schemas.BifrostResponsesResponse, *schemas.BifrostError) {
```

Comment:
you will need to write convertors for this HF -> Responses like you have done for chat completions, fallback is not the right approach here